perf/ring.go

```diff
@@ -98,13 +98,20 @@ func (ring *perfEventRing) Close() {
 	ring.mmap = nil
 }
 
-func createPerfEvent(cpu, watermark int, overwritable bool) (int, error) {
-	if watermark == 0 {
-		watermark = 1
+func createPerfEvent(cpu int, opts ReaderOptions) (int, error) {
+	wakeup := 0
+	bits := 0
```

Comment:
As the documentation for `WakeupEvents` states, it should be mutually exclusive with `Watermark`.
Can we add a check for the case where both, `WakeupEvents` and `Watermark`, are set  and return something like `fmt.Errorf("WakeupEvents and Watermark are mutually exclusive: %w", unix.EINVAL)`.

Reply:
Like this one? https://github.com/cilium/ebpf/pull/1404/files#diff-7b72e423487d9bde0d352acf2654d5e70f0c10898776b867fc3bac2b38f9ceafR199-R201

Reply:
ah - good point! placing the check in `NewReaderWithOptions()` before doing allocations make sense!